### PR TITLE
Bugfix/5045 : Modified the existing logic to comply crossref resolution with biblatex specification

### LINKED ISSUE #5045: crossref resolution does not comply with biblatex specification
**JabRef version** 

> JabRef 5.0-dev--snapshot--2019-06-10--master--eb42850f7
> Linux 4.4.0-150-generic amd64 
> Java 1.8.0_212

on

> Ubuntu 16.04

- [x] I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

**Steps to reproduce the behavior**

1. create a *Proceedings* entry `Foo2019`
```
@Proceedings{Foo2019,
  title  = {Foo Conference},
  date   = {2019},
}
```
2. create an *InProceedings* entry `Bar2019` with crossref to `Foo2019`
```
@Proceedings{Bar2019,
  title  = {Bar Paper},
  crossref = {Foo2019},
}
```
3. have a look at the `Bar2019` row at the main table view

**Observed behavior**

The *year* field of the `Bar2019` entry gets populated with the value from the `Foo2019` entry.

**Expected behavior**

All fileds (esp. *year* and *journal/booktitle*) get populated with values obtained from the crossref'ed entry.

**Explanation**

According to the [biblatex documentation](https://ftp.agdsn.de/pub/mirrors/latex/dante/macros/latex/contrib/biblatex/doc/biblatex.pdf) (Appendix B), the *booktitle* field of an *InProcessings* entry inherits the value from the *title* field of the referenced *Proceedings* entry.

### PR DESCRIPTION
**Pull request by CyraxSector**
Fixes  #5045

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Thanks for your contribution. Could you please add a test for the changed code?

**Comment by CyraxSector** [comment]
@tobiasdiez @Siedlerchr sure I'll make those amendments and the UTs

**Comment by CyraxSector** [comment]
@tobiasdiez Could you please review the latest commit. Thanks :)

**Comment by CyraxSector** [comment]
> Looks good to me now.
> Please merge the latest master changes in. There have been renaming of the Fields classes, so you need to fix this. Then we can merge

@Siedlerchr I pulled the changes from master. This PR can be merged to the master now. Thanks

File: src/main/java/org/jabref/model/entry/BibEntry.java [hunk 1]
```diff
@@ -21,6 +21,8 @@
 import javafx.collections.FXCollections;
 import javafx.collections.ObservableMap;
 
+import org.jabref.logic.importer.fetcher.CrossRef;
+import org.jabref.logic.importer.fileformat.bibtexml.Inproceedings;
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/model/entry/BibEntry.java]: Please check your imports, it seems like you have imported the type from the wrong package.
And CrossRef here looks also odd.

**Comment by CyraxSector** [comment]
> Please review the imports.

Fixed and organized imports. Please review and merge

**Comment by Siedlerchr** [comment]
> 
> 
> > Please review the imports.
> 
> Fixed and organized imports. Please review and merge

Please have a look at travis checkstyle output. Star imports are not allowed.
If you use Intellij you can directly import the code style from the config folder
For eclipse you can run ./gradlew eclipse to automatically import the correct code style

If you then run organize imports in the IDE it will be correct

**Comment by CyraxSector** [comment]
> > > Please review the imports.
> > 
> > 
> > Fixed and organized imports. Please review and merge
> 
> Please have a look at travis checkstyle output. Star imports are not allowed.
> If you use Intellij you can directly import the code style from the config folder
> For eclipse you can run ./gradlew eclipse to automatically import the correct code style
> 
> If you then run organize imports in the IDE it will be correct

Fixed imports issue. Please review

**Pull request merged**